common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -333,18 +334,29 @@ protected void reportInstancesLeak(String resourceType) {
     }
 
     @SuppressWarnings("deprecation")
-    private final class DefaultResourceLeak extends PhantomReference<Object> implements ResourceLeakTracker<T>,
-            ResourceLeak {
-        private final Record head;
+    private static final class DefaultResourceLeak<T>
+            extends PhantomReference<Object> implements ResourceLeakTracker<T>, ResourceLeak {
 
-        // This will be updated once a new Record will be created and added
-        private Record tail;
+        @SuppressWarnings("unchecked") // generics and updaters do not mix.
+        private static final AtomicReferenceFieldUpdater<DefaultResourceLeak<?>, Record> headUpdater =
+                (AtomicReferenceFieldUpdater)
+                        AtomicReferenceFieldUpdater.newUpdater(DefaultResourceLeak.class, Record.class, "head");
 
+        @SuppressWarnings("unchecked") // generics and updaters do not mix.
+        private static final AtomicIntegerFieldUpdater<DefaultResourceLeak<?>> droppedRecordsUpdater =
+                (AtomicIntegerFieldUpdater)
+                        AtomicIntegerFieldUpdater.newUpdater(DefaultResourceLeak.class, "droppedRecords");
+
+        private volatile Record head;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "head" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1068)